test-page/src/views/Product.vue

```diff
@@ -451,9 +390,145 @@ export default {
+      "associations[manufacturer][]": true,
+      "associations[children][associations][options][associations][group][]": true,
+    }
+    this.productResponse = await getProduct(this.$route.params.id, associations);
+    // get the missing info from parent product
+    if (this.productResponse.parentId) {
+      const { name, description, media, cover } = await getProduct(this.productResponse.parentId, {
+              "associations[media][]": true
+      })
+      this.productResponse.name = name;
+      this.productResponse.description = description
+      this.productResponse.media = media
+      this.productResponse.cover = cover
+    }
+  },
+  watch: {
+    selectedColor(value) {
+      this.redirectToSimple(value)
+    }
+  },
+  computed: {
+    product() {
+      return this.productResponse
+    },
+    price() {
+      return this.product.price ? this.product.price[0].gross : 0
+    },
+    name(){
+      return this.product.name
+    },
+    description() {
+      return this.product.description
+    },
+    ratingAverage(){
+      return this.product.ratingAverage
+    },
+    hasChildren() {
+      return this.product.childCount > 0
+    },
+    isSimple() {
+      return !!this.product.parentId
+    },
+    mainImage() {
+      return this.product.cover ? this.product.cover.media.url : ""
+    },
+    mediaGallery() {
+      return this.product && this.product.media ? this.product.media.map(media => 
+        {
+            const smallThumb = media.media.thumbnails.find(thumb => thumb.width == "400")
+            const normalThumb = media.media.thumbnails.find(thumb => thumb.width == "800")
+            const bigThumb = media.media.thumbnails.find(thumb => thumb.width == "1920")
+            return {
+              small: { url: smallThumb ? smallThumb.url : "" },
+              normal: { url: normalThumb ? normalThumb.url: "" },
+              big: { url: bigThumb ? bigThumb.url: "" }
+            }
+        }
+      ) : []
+    },
+    properties() {
+      if (!this.product || !this.product.properties) {
+        return []
+      }
+
+      const propertyList = this.product.properties.map(property => ({
+        name: property.group.name,
+        value: property.name
+      }))
+ 
+      return propertyList
+    },
+    color() {
+      if (!this.isSimple) {
+        return ""
+      }
+
+      return this.product.options.find(option => option.group.name === "color")
+    },
+    size() {
+      if (!this.isSimple) {
+        return ""
+      }
+
+      return this.product.options.find(option => option.group.name === "size")
+    },
+    colors() {
+      if (!this.product && !this.product.children) {
+        return []
+      }
+      
+      return this.extractOptions("color")
+    },
+    sizes() {
+      if (!this.product && !this.product.children) {
+        return []
+      }
+
+      return this.extractOptions("size")
+    },
+    reviews() {
+      return this.product.productReviews
```

Comment:
you're checking `reviews.length` much, so better ensure it always is at least an empty array
```suggestion
      return this.product.productReviews || []
```